src/sqlfluff/core/rules/base.py

```diff
@@ -332,6 +373,31 @@ def get_fix_slices(
             # We return an empty set because this edit doesn't touch anything
             # in the source.
             return set()
+        elif (
+            self.edit_type == "replace"
+            and all(edit.is_type("raw") for edit in cast(List[RawSegment], self.edit))
+            and all(edit._source_fixes for edit in cast(List[RawSegment], self.edit))
+        ):
+            # This is just source fixes. We can go directly to the touched raw
+            # slices and only if they overlap directly.
```

Comment:
Good question 🤔 - I'm not sure what I meant here. I think this might be an outdated comment. I'll revise it.

Reply:
Yeah no idea. I've updated it to something more descriptive of what's going on:
```python
# As an exception to the general rule about "replace" fixes (where
# they're only safe if they don't touch a templated section at all),
# source-only fixes are different. This clause handles that exception.

# So long as the fix is *purely* source-only we can assume that the
# rule has done the relevant due diligence on what it's editing in
# the source and just yield the source slices directly.

# More complicated fixes that are a blend or source and templated
# fixes are currently not supported but this (mostly because they've
# not arisen yet!), so further work would be required to support them
# elegantly.
```